source/docs/software/commandbased/commands-v3/what-is-command-based-v3.rst

```diff
@@ -0,0 +1,241 @@
+# What Is Commands v3?
+
+Commands v3 is WPILib's next-generation command-based framework for Java. It introduces imperative-style command writing using coroutines, allowing you to write robot behaviors as straightforward sequential code instead of complex chains of lambda expressions and decorators.
+
+.. warning::
+   Commands v3 requires **Java 21 or later** and is **Java-only**. If you need C++, Python, or earlier Java versions, use :ref:`docs/software/commandbased/commands-v2/index:Commands v2 Programming`.
+
+## Command-Based Programming Philosophy
+
+Like Commands v2, Commands v3 is a :term:`design pattern` for organizing robot software. It's not the only way to write a robot program, but it's effective for creating clean, maintainable, and reusable code.
```

Comment:
I don't think the v3 intro needs to include comparisons with v2; that's better done in the migration guides